### ISSUE TITLE
Page List block inside navigation block will result invalid html

### ISSUE THREAD

**Issue by saulirajala** [WordPress/gutenberg#60500]
### Description

`page-list` -block has features like `showSubmenuIcon` and `openSubmenusOnClick` that comes from the context provided by `navigation` -block, so I would assume that it's completely valid to use page-list block inside of navigation block. But doing so, will result invalid html since both navigation and page-list blocks starts with `<ul>`-element. So we would have `<ul><ul><li>`… structure, which is [not allowed](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul#technical_summary)

### Step-by-step reproduction instructions

1. Add navigation block
2. Add page-list block as inner block of navigation
3. check the html tags

The live demo of this issue is in WordPress Playground: https://playground.wordpress.net/. Inspect the navigation in header. 

### Screenshots, screen recording, code snippet

![Screenshot 2024-04-05 at 11 08 35](https://github.com/WordPress/gutenberg/assets/5536354/4fdbe07c-da82-47d8-82e7-e18a3c865e3c)

### Environment info

- WordPress 6.5
- Twenty Twenty-Four

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by saulirajala** [comment]
The issue still exists. Even with the latest Gutenberg plugin installed. You can verify it via WordPress Playground https://playground.wordpress.net/?plugin=gutenberg

**Comment by saulirajala** [comment]
I did [PR](https://github.com/WordPress/gutenberg/pull/61475) that fixes this. You can test it in [Playground PR Reviewer](https://playground.wordpress.net/#%7B%22landingPage%22:%22/%22,%22steps%22:%5B%7B%22step%22:%22login%22,%22username%22:%22admin%22,%22password%22:%22password%22%7D,%7B%22step%22:%22mkdir%22,%22path%22:%22/wordpress/pr%22%7D,%7B%22step%22:%22writeFile%22,%22path%22:%22/wordpress/pr/pr.zip%22,%22data%22:%7B%22resource%22:%22url%22,%22url%22:%22/plugin-proxy.php?org=WordPress&repo=gutenberg&workflow=Build%2520Gutenberg%2520Plugin%2520Zip&artifact=gutenberg-plugin&pr=61475%22,%22caption%22:%22Downloading%20Gutenberg%20PR%2061475%22%7D,%22progress%22:%7B%22weight%22:2,%22caption%22:%22Applying%20Gutenberg%20PR%2061475%22%7D%7D,%7B%22step%22:%22unzip%22,%22zipPath%22:%22/wordpress/pr/pr.zip%22,%22extractToPath%22:%22/wordpress/pr%22%7D,%7B%22step%22:%22installPlugin%22,%22pluginZipFile%22:%7B%22resource%22:%22vfs%22,%22path%22:%22/wordpress/pr/gutenberg.zip%22%7D%7D%5D%7D) 
![Screenshot 2024-05-08 at 11 00 50](https://github.com/WordPress/gutenberg/assets/5536354/067b4689-46ce-432e-a46e-5c830ba32413)

**Comment by saulirajala** [comment]
Actually my solution will introduce another issue: the `ul.wp-block-page-list` -element&class is missing from DOM, which is not good. So we need to think another way to fix this.

**Comment by saulirajala** [comment]
Seems like `core/navigation` -block already has functionality to wrap inner block with `<li>`, but `core/page-list` -block was just missing from `$needs_list_item_wrapper` -list. [Added it](https://github.com/WordPress/gutenberg/pull/61482) there, which seems to solve the issue. You can test the PR in [Gutenberg PR reviewer](https://playground.wordpress.net/#%7B%22landingPage%22:%22/%22,%22steps%22:%5B%7B%22step%22:%22login%22,%22username%22:%22admin%22,%22password%22:%22password%22%7D,%7B%22step%22:%22mkdir%22,%22path%22:%22/wordpress/pr%22%7D,%7B%22step%22:%22writeFile%22,%22path%22:%22/wordpress/pr/pr.zip%22,%22data%22:%7B%22resource%22:%22url%22,%22url%22:%22/plugin-proxy.php?org=WordPress&repo=gutenberg&workflow=Build%2520Gutenberg%2520Plugin%2520Zip&artifact=gutenberg-plugin&pr=61482%22,%22caption%22:%22Downloading%20Gutenberg%20PR%2061482%22%7D,%22progress%22:%7B%22weight%22:2,%22caption%22:%22Applying%20Gutenberg%20PR%2061482%22%7D%7D,%7B%22step%22:%22unzip%22,%22zipPath%22:%22/wordpress/pr/pr.zip%22,%22extractToPath%22:%22/wordpress/pr%22%7D,%7B%22step%22:%22installPlugin%22,%22pluginZipFile%22:%7B%22resource%22:%22vfs%22,%22path%22:%22/wordpress/pr/gutenberg.zip%22%7D%7D%5D%7D)

**Comment by saulirajala** [comment]
This is still an issue even with WordPress 6.7 RC2 and the new twentytwentyfive theme. You can test the issue in [RC2 playground](https://playground.wordpress.net/#%7B%20%22preferredVersions%22:%20%7B%20%22php%22:%20%228.0%22,%20%22wp%22:%20%22https://playground.wordpress.net/plugin-proxy.php?build-ref=5087acab42d5354c8766303d13b78c8a77ac331b%22%20%7D,%20%22features%22:%20%7B%20%22networking%22:%20true%20%7D,%20%22steps%22:%20%5B%20%7B%20%22step%22:%20%22login%22,%20%22username%22:%20%22admin%22,%20%22password%22:%20%22password%22%20%7D,%20%7B%20%22step%22:%20%22importFile%22,%20%22file%22:%20%7B%20%22resource%22:%20%22url%22,%20%22url%22:%20%22https://raw.githubusercontent.com/wpaccessibility/a11y-theme-unit-test/master/a11y-theme-unit-test-data.xml%22%20%7D%20%7D,%20%7B%20%22step%22:%20%22importFile%22,%20%22file%22:%20%7B%20%22resource%22:%20%22url%22,%20%22url%22:%20%22https://raw.githubusercontent.com/WordPress/theme-test-data/master/themeunittestdata.wordpress.xml%22%20%7D%20%7D,%20%7B%20%22step%22:%20%22installPlugin%22,%20%22pluginZipFile%22:%20%7B%20%22resource%22:%20%22wordpress.org/plugins%22,%20%22slug%22:%20%22query-monitor%22%20%7D,%20%22options%22:%20%7B%20%22activate%22:%20false%20%7D,%20%22progress%22:%20%7B%20%22weight%22:%202%20%7D%20%7D,%20%7B%20%22step%22:%20%22installPlugin%22,%20%22pluginZipFile%22:%20%7B%20%22resource%22:%20%22wordpress.org/plugins%22,%20%22slug%22:%20%22create-block-theme%22%20%7D,%20%22progress%22:%20%7B%20%22weight%22:%202%20%7D%20%7D,%20%7B%20%22step%22:%20%22installPlugin%22,%20%22pluginZipFile%22:%20%7B%20%22resource%22:%20%22wordpress.org/plugins%22,%20%22slug%22:%20%22debug-bar%22%20%7D,%20%22progress%22:%20%7B%20%22weight%22:%202%20%7D%20%7D,%20%7B%20%22step%22:%20%22installPlugin%22,%20%22pluginZipFile%22:%20%7B%20%22resource%22:%20%22wordpress.org/plugins%22,%20%22slug%22:%20%22health-check%22%20%7D,%20%22progress%22:%20%7B%20%22weight%22:%202%20%7D%20%7D,%20%7B%20%22step%22:%20%22installPlugin%22,%20%22pluginZipFile%22:%20%7B%20%22resource%22:%20%22wordpress.org/plugins%22,%20%22slug%22:%20%22test-reports%22%20%7D,%20%22progress%22:%20%7B%20%22weight%22:%202%20%7D%20%7D,%20%7B%20%22step%22:%20%22installPlugin%22,%20%22pluginZipFile%22:%20%7B%20%22resource%22:%20%22wordpress.org/plugins%22,%20%22slug%22:%20%22user-switching%22%20%7D,%20%22progress%22:%20%7B%20%22weight%22:%202%20%7D%20%7D%20%5D%20%7D). I updated [the pr](https://github.com/WordPress/gutenberg/pull/61482) since it had merge conflicts. 

Any change to get this fixed before WP6.7 release?

**Comment by nick-novak** [comment]
Is there any update on this? We like using the page list feature when our editors build out their navigations. But the invalid HTML stops us from doing this.

**Comment by SirLouen** [comment]
## Reproduction Report
### Description
✅ This report validates that the issue can be reproduced.

### Environment
- WordPress: 6.9-alpha-60093-src
- PHP: 8.2.29
- Server: Apache/2.4.62 (Debian)
- Database: mysqli (Server: 11.8.2-MariaDB-ubu2404 / Client: mysqlnd 8.2.29)
- Browser: Chrome 138.0.0.0
- OS: Windows 10/11
- Theme: Twenty Twenty-Five 1.3
- MU Plugins: None activated
- Plugins:
  * Gutenberg 21.3.0-rc.1
  * Test Reports 1.2.0

### Test Instructions
1. Go to Appearance > Editor and go into the Editor
3. Select the Navigation block inside the header
3. Add a Page List 
4. Save and go to the Front End
5. Check the HTML code for the Navigation Block
6. 🐞The structure is as follows: `<ul><ul><li>` and it should be `<ul><li><ul><li>`

### Actual Results
1.  ✅ Error condition occurs (reproduced).

### Additional Notes
- It seems that it's already being sorted here #61482. But it appears that it has been stalling for almost one year now. 
- So it's time to propose another patch. In case anyone wants to take it, [the idea provided](https://github.com/WordPress/gutenberg/pull/61482#issuecomment-2496965766) by @talldan could be a good starting point.

### Supplemental Artifacts
<img width="458" height="307" alt="Image" src="https://github.com/user-attachments/assets/7f16815e-08ea-4a6e-b71f-004a6b02cc65" />

**Comment by SirLouen** [comment]
It seems it's not an easy fix as it seems, and currently all the proposed solutions are under deliberation in the PR https://github.com/WordPress/gutenberg/pull/61482

**Comment by saulirajala** [comment]
I picked this back up and set up a test on WordPress 7.0 with Twenty Twenty-Five, comparing the current behavior against the wrapper-omission fix @talldan suggested (dropping the Page List's own `<ul>`), to see exactly what that fix costs. It turns out this is a design trade-off that needs a maintainer decision, not just a patch, so I want to lay it out before writing more code.

## The core problem: you can only get two of these three

For a Page List that is a direct child of a Navigation block, at most two of these can be true at once:

1. Valid HTML (no `<ul>` directly inside another `<ul>`)
2. Correct flex layout (page items behave as individual nav items)
3. A styleable container (block-support styles and `theme.json` `styles.blocks.core/page-list` keep working)

Current Core code keeps (2) and (3) but loses (1): the Page List renders its own `<ul class="wp-block-page-list …">` inside the Navigation `<ul>`, which is invalid.

The extra-`<li>`-wrapper approach attempted in #61482 keeps (1) and (3) but loses (2). It wraps the list in an `<li>` at the Navigation level (adding `core/page-list` to the block's list-item-wrapper set), so the `<ul class="wp-block-page-list">` and its styles stay intact, but all pages then collapse into a single flex item and theme layouts break.

The wrapper-omission approach @talldan suggested keeps (1) and (2) but loses (3): there is no longer any element carrying the block's styling.

One alternative that might come to mind, `display:contents` on an `<li>` wrapper, doesn't rescue it either. It satisfies (1) and (2) but removes the box, so background, border and padding still don't render, so it still loses (3).

## What the wrapper-omission fix actually costs

I tested the wrapper-omission approach (omit the `<ul>` when Page List is a direct child of Navigation, and move the block-support classes and inline styles onto the top-level `<li>` items) and compared it against WP 7.0.

### Scenario A, text color from `theme.json` global styles plus a per-instance font size:

**BEFORE `<ul><ul>`, invalid, but the global pink text and the larger font both apply**

<img width="1200" height="815" alt="Image" src="https://github.com/user-attachments/assets/627c3b0b-f2c1-421e-9c8d-067cd8df7f85" />

**AFTER `<ul><li>`, valid, font size survives, but the global text color is gone**

<img width="1200" height="815" alt="Image" src="https://github.com/user-attachments/assets/1f58de87-2db3-4ec0-b1ec-a274c85aab13" />

### Scenario B, per-instance background, border and padding:

**BEFORE: one box around the whole list**

<img width="1200" height="935" alt="Image" src="https://github.com/user-attachments/assets/abc87122-032f-4def-8de4-463cb8757915" />


**AFTER: the box, background and padding repeat on every item**

<img width="1200" height="983" alt="Image" src="https://github.com/user-attachments/assets/bfcd0c5a-780a-499d-96b5-3c992a501e88" />



### What survives and what breaks

| Style type | Behavior after wrapper omission |
| --- | --- |
| Inheriting (text color, font size, font family, line-height) | Identical. Moved to each `<li>` and inherited by the link. |
| Box-model, per-instance (background, border, padding, margin, radius) | Visually broken. One container box becomes one box per item. |
| Global `theme.json` (`styles.blocks.core/page-list`) | Gone. The `.wp-block-page-list` selector still ships in the stylesheet but matches no element. |

So the idea that moving styles to the `<li>` items keeps styling working is only partly true. It holds for typography and color, not for box-model or global styles. And since Page List inside Navigation is one of the most common placements, the global-styles loss isn't a niche edge case.

This is the same editor/front-end markup divergence tracked in #61170, and the style-loss half is tracked in #80000 (it already affects the Submenu case today, and the wrapper-omission fix would extend it to the Navigation-child case).

## The question for maintainers

It seems that you can't get all three, so which trade-off do we want?

1. Ship wrapper omission, and accept that global and box-model Page List styling in Navigation stops working, in exchange for valid markup.
2. Keep the container somehow, via `block.json` `selectors` or another mechanism that survives
3. Something else, for example rethinking how Page List participates in Navigation markup (relates to #61170).

I'd rather get a steer on direction than push a PR that bakes in a regression. cc @getdave @talldan @SirLouen @mcsf (who recently refactored this area in #79048).
